proxy/proxy_test.go

```diff
@@ -1332,6 +1332,289 @@ var _ = Describe("Proxy", func() {
+						g.Expect(logStr).To(MatchRegexp("response_time:4"))        // makes 4 seconds total response time
+						g.Expect(logStr).To(MatchRegexp("gorouter_time:0"))
+						g.Expect(logStr).To(MatchRegexp("failed_attempts:2"))
+						g.Expect(logStr).To(MatchRegexp("app_index:\"3\""))
+
+					}, "20s").Should(Succeed()) // we don't know which endpoint will be chosen first, so we have to try until sequence 1,2,3 has been hit
+				})
+			})
+		})
+
+		Context("lookup errors when attempt details logging is enabled", func() {
+			BeforeEach(func() {
+				conf.Logging.EnableAttemptsDetails = true
+			})
+
+			It("logs no backend_time on missing app route", func() {
+				conn := dialProxy(proxyServer)
+
+				req := test_util.NewRequest("GET", "does-not-exist", "/", nil)
+
+				started := time.Now()
+				conn.WriteRequest(req)
+
+				resp, err := http.ReadResponse(conn.Reader, &http.Request{})
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(resp.StatusCode).To(Equal(http.StatusNotFound))
+				Expect(time.Since(started)).To(BeNumerically("<", 5*time.Second))
+
+				Eventually(func() (int64, error) {
+					fi, err := f.Stat()
+					if err != nil {
+						return 0, err
+					}
+					return fi.Size(), nil
+				}, "5s").ShouldNot(BeZero())
+
+				logBytes, err := os.ReadFile(f.Name())
+				Expect(err).NotTo(HaveOccurred())
+				logStr := string(logBytes)
+				Expect(strings.HasPrefix(logStr, "does-not-exist - [")).To(BeTrue())
+				Expect(logStr).To(ContainSubstring(`response_time:`))
+				Expect(logStr).To(ContainSubstring(`gorouter_time:`))
+				Expect(logStr).To(MatchRegexp("response_time:0"))
+				Expect(logStr).To(MatchRegexp("gorouter_time:0"))
+				Expect(logStr).To(MatchRegexp("backend_time:\"-\""))
+			})
+			It("logs no backend_time on invalid X-CF-App-Instance header", func() {
+				conn := dialProxy(proxyServer)
+
+				req := test_util.NewRequest("GET", "invalid-app-instance-header", "/", nil)
+				req.Header.Set("X-CF-App-Instance", "invalid-instance")
+				started := time.Now()
+				conn.WriteRequest(req)
+
+				resp, err := http.ReadResponse(conn.Reader, &http.Request{})
+				Expect(err).NotTo(HaveOccurred())
+
+				Expect(resp.StatusCode).To(Equal(http.StatusBadRequest))
+				Expect(time.Since(started)).To(BeNumerically("<", 5*time.Second))
+
+				Eventually(func() (int64, error) {
+					fi, err := f.Stat()
+					if err != nil {
+						return 0, err
+					}
+					return fi.Size(), nil
+				}, "5s").ShouldNot(BeZero())
+
+				logBytes, err := os.ReadFile(f.Name())
+				Expect(err).NotTo(HaveOccurred())
+				logStr := string(logBytes)
+				Expect(strings.HasPrefix(logStr, "invalid-app-instance-header - [")).To(BeTrue())
+				Expect(logStr).To(ContainSubstring(`response_time:`))
+				Expect(logStr).To(ContainSubstring(`gorouter_time:`))
+				Expect(logStr).To(MatchRegexp("response_time:0"))
+				Expect(logStr).To(MatchRegexp("gorouter_time:0"))
+				Expect(logStr).To(MatchRegexp("backend_time:\"-\""))
```

Comment:
here, and wherever we have the escaped `\"`, we could use the literal string with backticks...